File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/mastery/MasteryDao.java
package com.legue.axel.lolsummonertool.database.dao.mastery;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.legue.axel.lolsummonertool.database.model.mastery.Mastery;

import java.util.List;

@Dao
public interface MasteryDao {

    @Query("SELECT * FROM masteries ORDER BY id")
    LiveData<List<Mastery>> getMasteries();

    @Query("SELECT * FROM masteries WHERE id = :masteryId")
    LiveData<Mastery> getMasteryById(int masteryId);

    @Insert
    void insertMastery(Mastery mastery);

    @Insert
    void insertAllMasteries(List<Mastery> masteries);

    @Delete
    void deleteMastery(Mastery mastery);

    @Query("DELETE FROM masteries")
    void deleteAll();

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateMastery(Mastery mastery);
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/mastery/MasteryDao.kt
package com.legue.axel.lolsummonertool.database.dao.mastery

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update

import com.legue.axel.lolsummonertool.database.model.mastery.Mastery

@Dao
interface MasteryDao {

    @Query("SELECT * FROM masteries ORDER BY id")
    fun getMasteries(): LiveData<List<Mastery>>

    @Query("SELECT * FROM masteries WHERE id = :masteryId")
    fun getMasteryById(masteryId: Int): LiveData<Mastery>

    @Insert
    fun insertMastery(mastery: Mastery)

    @Insert
    fun insertAllMasteries(masteries: List<Mastery>)

    @Delete
    fun deleteMastery(mastery: Mastery)

    @Query("DELETE FROM masteries")
    fun deleteAll()

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateMastery(mastery: Mastery)
}
